File: Roman arabic numerals/roman-arabic-numerals.cs
﻿using System;

namespace Base
{
    class Program
    {

        public static int RomArab(string p)
        {
            int z = 0;
            try
            {
                for(int i = 0; i < p.Length; i ++)
                {
                    if(char.ToLower(p[i]) == 'i')
                    {
                        try
                        {
                            if(char.ToLower(p[i+1]) == 'v' || char.ToLower(p[i+1]) == 'x')
                            { 
                                z -= 1;
                            }
                            else
                            {
                                z += 1;
                            }
                        }
                        catch
                        {
                            z += 1;
                        }
                    }
                    else if(char.ToLower(p[i]) == 'v')
                    {
                        z += 5;
                    }
                    else if(char.ToLower(p[i]) == 'x')
                    {
                        try
                        {
                            if(char.ToLower(p[i+1]) == 'c' || char.ToLower(p[i+1]) == 'l')
                            {
                                z -= 10;
                            }
                            else
                            {
                                z += 10;
                            }
                        }
                        catch
                        {
                            z += 10;
                        }
                    }
                    else if(char.ToLower(p[i]) == 'l')
                    {
                        z += 50;
                    }
                    else if(char.ToLower(p[i]) == 'c')
                    {
                        try
                        {
                            if(char.ToLower(p[i+1]) == 'm' || char.ToLower(p[i+1]) == 'd')
                            { 
                                z -= 100;
                            }
                            else
                            {
                                z += 100;
                            }
                        }
                        catch
                        {
                            z += 100;
                        }
                    }
                    else if(char.ToLower(p[i]) == 'd')
                    {
                        z += 500;
                    }
                    else if(char.ToLower(p[i]) == 'm')
                    {
                        z += 1000;
                    }
                    else
                    {
                        Console.WriteLine("Uncorrect input");
                        return 0;
                    }
                }
            }
            catch
            {
                Console.WriteLine("Uncorrect input");
                return 0;
            }
            return z;
        }

        public static string ArabRom(int s)
        {
            string v = "";
            try
            {
                while(s > 0)
                {
                    if(s >= 1000)
                    {
                        s -= 1000;
                        v += "M";
                    }
                    else if(s >= 900)
                    {
                        s -= 900;
                        v += "CM";
                    }
                    else if(s >= 500)
                    {
                        s -= 500;
                        v += "D";
                    }
                    else if(s >= 400)
                    {
                        s -= 400;
                        v += "CD";
                    }
                    else if(s >= 100)
                    {
                        s -= 100;
                        v += "C";
                    }
                    else if(s >= 90)
                    {
                        s -= 90;
                        v += "XC";
                    }
                    else if(s >= 50)
                    {
                        s -= 50;
                        v += "L";
                    }
                    else if(s >= 40)
                    {
                        s -= 40;
                        v += "XL";
                    }
                    else if(s >= 10)
                    {
                        s -= 10;
                        v += "X";
                    }
                    else if(s >= 9)
                    {
                        s -= 9;
                        v += "IX";
                    }
                    else if(s >= 5)
                    {
                        s -= 5;
                        v += "V";
                    }
                    else if(s >= 4)
                    {
                        s -= 4;
                        v += "IV";
                    }
                    else if(s >= 1)
                    {
                        s -= 1;
                        v += "I";
                    }
                }
            }
            catch
            {
                Console.WriteLine("Uncorrect input");
                return "";
            }
            return v;
        }

        public static void Main()
        {
            Console.ReadLine();
        }
    }
}

File: Roman arabic numerals/roman-arabic-numerals.py
def rom_arab(p):
    z = 0
    try:
        p = str(p).lower()
        for i in range(0, len(p)):
            if p[i] == 'i':
                try:
                    if p[i+1] == 'v' or p[i+1] == 'x': 
                        z -= 1
                    else:
                        z += 1
                except:
                    z += 1
            elif p[i] == 'v':
                z += 5
            elif p[i] == 'x':
                try:
                    if p[i+1] == 'c' or p[i+1] == 'l': 
                        z -= 10
                    else:
                        z += 10
                except:
                    z += 10
            elif p[i] == 'l':
                z += 50
            elif p[i] == 'c':
                try:
                    if p[i+1] == 'm' or p[i+1] == 'd': 
                        z -= 100
                    else:
                        z += 100
                except:
                    z += 100
            elif p[i] == 'd':
                z += 500
            elif p[i] == 'm':
                z += 1000
            else:
                return None
    except:
        return None
    return z

def arab_rom(s):
    v = ""
    try:
        s = int(s)
        while s > 0:
            if s >= 1000:
                s -= 1000
                v += "M"
            elif s >= 900:
                s -= 900
                v += "CM"
            elif s >= 500:
                s -= 500
                v += "D"
            elif s >= 400:
                s -= 400
                v += "CD"
            elif s >= 100:
                s -= 100
                v += "C"
            elif s >= 90:
                s -= 90
                v += "XC"
            elif s >= 50:
                s -= 50
                v += "L"
            elif s >= 40:
                s -= 40
                v += "XL"
            elif s >= 10:
                s -= 10
                v += "X"
            elif s >= 9:
                s -= 9 
                v += "IX"
            elif s >= 5:
                s -= 5
                v += "V"
            elif s >= 4:
                s -= 4
                v += "IV"
            elif s >= 1:
                s -= 1
                v += "I"
    except:
        return None
    return v